plugins/GeoIp2/GeoIP2AutoUpdater.php

```diff
@@ -741,4 +744,35 @@ public static function isDbIpUrl($url)
     {
         return !! preg_match('/db-ip\.com/', $url);
     }
+
+    protected static function isPaidDbIpUrl($url)
+    {
+        return !! preg_match('/db-ip\.com\/account\/[0-9a-z]+\/db/', $url);
+    }
+
+    protected function fetchPaidDbIpUrl($url)
+    {
+        $content = trim($this->fetchUrl($url));
+
+        if (0 === strpos($content, 'http')) {
+            return $content;
+        }
+
+        $content = json_decode($content, true);
+
+        if (!empty($content['mmdb']['url'])) {
+            return $content['mmdb']['url'];
+        }
+
+        if (!empty($content['url'])) {
+            return $content['url'];
+        }
+
+        throw new Exception('Unable to determine download url');
+    }
+
+    protected function fetchUrl($url)
+    {
+        return Http::fetchRemoteFile($url);
```

Comment:
@sgiehl just realising we should be checking whether internet is enabled and if not, throw an exception maybe? 

BTW out of curiosity: Any reason using `fetchRemoteFile` over `sendHttpRequest`?

Reply:
Checking if internet is enabled is done in `__construct`, so need to do that another time.

> BTW out of curiosity: Any reason using fetchRemoteFile over sendHttpRequest?

No. Actually not. Feel free to change that...